### PR TITLE
Add c2c modules to distribution

### PR DESCRIPTION
**Pull request by anuruddhal**
## Purpose
- Add c2c modules to distribution

### CODE REVIEW

File: ballerina/build.gradle [hunk 1]
```diff
@@ -203,6 +214,20 @@ task copyOtherRepos(type: Copy) {
...
+        def artifactExtractedPath = "${buildDir}/target/extracted-distributions/" + artifact.name + "-zip"
+        into("bre/lib/") {
+            from "${artifactExtractedPath}/libs"
+        }
+        into("docs/") {
+            from "${artifactExtractedPath}/docs"
+        }
+        into("bir-cache/") {
+            from "${artifactExtractedPath}/caches/bir"
+        }
```
Comment (BuddhiWathsala) [inline ballerina/build.gradle:229]: Shouldn't we need to pack these C2C library artifacts to the new packerina cache?.
```gradle
// pack to new cache
into("cache/bir/") {
    from "${artifactExtractedPath}/caches/bir"
}
into("cache/balo/") {
    from "${artifactExtractedPath}/caches/balo"
}
into("cache/jar/") {
    from "${artifactExtractedPath}/caches/jar"
}
```

@anuruddhal @pramodya1994 @hevayo

Comment (anuruddhal) [reply ballerina/build.gradle:229]: No. 
The annotation processors work by copying `bir-cache` and `libs` only.  
Docker and k8s annotation processors do the same.

Comment (BuddhiWathsala) [reply ballerina/build.gradle:229]: Yes, but AFAIK the new packerina implementation is going to get rid of existing caches and move on to the new cache. I just want to confirm it.

Comment (hevayo) [reply ballerina/build.gradle:229]: @BuddhiWathsala You are right as per our plan we have to pack it there. But with the new package API change we need to revisit this so lets merge it.

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**